src/ios/FirebaseAnalyticsPlugin.m

```diff
@@ -143,6 +143,58 @@ - (void)showTrackingAuthorizationPopup:(CDVInvokedUrlCommand *)command {
     [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
 }
 
+- (void)setConsent:(CDVInvokedUrlCommand*)command
+{
+    CDVPluginResult* pluginResult = nil;
+    
+    NSString *jsonString = [command.arguments objectAtIndex:0];
+    NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error = nil;
+    NSArray *array = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
+    
+    if (error != nil) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                                        messageAsString:error.description];
+    } else {
+        NSMutableSet *seenTypes = [NSMutableSet set];
+        NSMutableDictionary *firebaseConsentDict = [NSMutableDictionary dictionary];
+        BOOL duplicatesFound = NO;
```

Comment:
Do we need this variable? Should we continue to iterate through the array after finding the first duplicated value?